src/plugins/intel_gna/src/gna_plugin.cpp

```diff
@@ -1429,18 +1229,33 @@ RequestStatus GNAPlugin::WaitFor(uint32_t request_idx, int64_t millisTimeout) {
                                         transpose_output_info->second);
         }
 
-        ExportScores(outputBlob->buffer(),
```

Comment:
So this change caused some problems with InferenceEngineUnitTests which I've been fighting for some time now.
Before, scores were exported as int32 (without scaling), and only if gnadevice was true the UnscaleAndCast function was called, which performed scaling and conversion to the target precision. This caused looping over scores two times instead of one and duplicated a lot of code.
Here, I've simplified it to do unscale and cast in one loop, in ExportScores. This doesn't change the behavior for gnadevice==true (apart from improved performance), but for gnadevice==false we don't have an option to not scale anymore - so I've simply put scale factor of 1.0f here. 
This additional division by 1.0f causes discrepancies between expected and actual values in a few of the mentioned tests, usually on 6th decimal place (eg. expected 0.14366889, got 0.14366913). I don't quite understand how a division by 1 (even taking into account that this is floating point, so the results may not be accurate) could impact the results so much, but apparently that's what happening

I see two possible solutions here:
- Change the expected values inside the unit tests. This would be the easier solution, but I don't know if it's acceptable. In one of the tests, the expected value isn't hard-coded but calculated on the fly from the input data, so change here would be more complicated but still relatively easy
- Don't perform division by 1.0f at all when gnadevice==false. This would mean that I'd have to add an additional argument to ExportScores, UnscaleTransposeAndCast as well as all of the AVX2-optimized functions to not perform the division. This solution, while complying fully with the previous behavior (I've tested it already), would significantly clutter the code, in my opinion.

I've temporarily reduced required precision of these tests to 4th decimal place to see if the PR will finaly pass on Windows CI (it did), but it's not an acceptable solution, obviously.

I'm not sure if I'm missing something here, would be great if someone could take a look at this.

Reply:
> So this change caused some problems with InferenceEngineUnitTests which I've been fighting for some time now.

This PR is intended to improve performance, so we should avoid any functional changes. I think threshold change is not accaptable. Also devision by 1.0f has no sense, so your second proposal is preferrable.